app/src/main/java/com/example/groceryshop/activities/data/DatabaseHelper.java

```diff
@@ -0,0 +1,315 @@
+        List<CategoryEntity> categoryEntityList = new ArrayList<>();
+        SQLiteDatabase dbCategory = getReadableDatabase();
+        Cursor cursor = dbCategory.rawQuery("SELECT * from category", null);
+        if (cursor != null && cursor.getCount() > 0) {
+            cursor.moveToFirst();
+            while (!cursor.isAfterLast()) {
+                CategoryEntity category = new CategoryEntity();
+                category.setIdCategory(cursor.getInt(cursor.getColumnIndex("idCategory")));
+                category.setImgCategory(cursor.getString(cursor.getColumnIndex("imgCategory")));
+                category.setNameCategory(cursor.getString(cursor.getColumnIndex("nameCategory")));
+                categoryEntityList.add(category);
+                cursor.moveToNext();
+            }
+            cursor.close();
+        }
+        return categoryEntityList;
+    }
+
+//GET USER
+    //singup
+
+    public long insertUser(UserEntity userEntity) {
+        sqLiteDatabase = context.openOrCreateDatabase(DB_NAME, Context.MODE_PRIVATE, null);
+        long newRow = 0;
+        ContentValues values = new ContentValues();
+        values.put("email", userEntity.email);
+        values.put("password", userEntity.passwordUser);
+        values.put("name", userEntity.fullName);
+        Toast.makeText(context, R.string.lbl_SignUpSuccess, Toast.LENGTH_SHORT).show();
+        newRow = sqLiteDatabase.insert("user", null, values);
+
+        return newRow;
+    }
+
+
+    private UserEntity dataUser(Cursor cursor) {
+        UserEntity userEntity1 = new UserEntity();
+            userEntity1.idUser = cursor.getInt(cursor.getColumnIndex("id"));
+            userEntity1.email = cursor.getString(cursor.getColumnIndex("email"));
+            userEntity1.passwordUser = cursor.getString(cursor.getColumnIndex("password"));
+            userEntity1.fullName = cursor.getString(cursor.getColumnIndex("name"));
+
+        return userEntity1;
+    }
+
+    public UserEntity getUserLogin(String email) {
+        SQLiteDatabase dbUser = getReadableDatabase();
+        Cursor cursor = dbUser.rawQuery(" SELECT * from user WHERE email= '" + email + "'", null);
+        UserEntity userEntity1 = new UserEntity();
+        if (cursor != null && cursor.moveToFirst()) {
+            userEntity1 = dataUser(cursor);
+            Log.e(TAG, "getUserLogin: " + userEntity1.email);
+        } return userEntity1;
+    }
+
+
+    public boolean checkEmail(String email1) {
+        SQLiteDatabase db = getReadableDatabase();
+        Cursor cursor = db.rawQuery(" SELECT email from user WHERE email= '" + email1 + "'", null);
+        if (cursor != null & cursor.moveToFirst()) {
+            String email2 = cursor.getString(cursor.getColumnIndex("email"));
+            Log.e(TAG, "checkEmail: " + email2);
+            if (email2.equals(email1)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public UserEntity Login(UserEntity userEntity) {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.